server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1409,6 +1410,23 @@ public ReplicationCheckpoint getLatestReplicationCheckpoint() {
         if (getEngineOrNull() == null) {
             return ReplicationCheckpoint.empty(shardId);
         }
+        // If index is closed, read replication checkpoint from last committed segments info on local store
+        if (indexSettings.getIndexMetadata().getState() == IndexMetadata.State.CLOSE) {
+            try {
+                final SegmentInfos segmentInfos = store.readLastCommittedSegmentsInfo();
```

Comment:
How will this behave if the engine is not yet open? I think we need to check for the existence of a commit first. This method will throw if one is not present.